packages/common/src/workflow-options.ts

```diff
@@ -39,13 +39,57 @@ export enum WorkflowIdReusePolicy {
 
   /**
    * Terminate the current workflow if one is already running.
+   *
+   * This option belongs in `WorkflowIdConflictPolicy` but it is here for backwards compatibility.
+   * With no Running Workflow, the behavior is the same as `WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE`,
+   * but with a Running Workflow, it behaves like `WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING`.
+   *
+   * *Note: When setting this option, `WorkflowIdConflictPolicy` should have value `WORKFLOW_ID_CONFLICT_POLICY_UNSPECIFIED`*
    */
   WORKFLOW_ID_REUSE_POLICY_TERMINATE_IF_RUNNING = 4,
 }
 
 checkExtends<temporal.api.enums.v1.WorkflowIdReusePolicy, WorkflowIdReusePolicy>();
 checkExtends<WorkflowIdReusePolicy, temporal.api.enums.v1.WorkflowIdReusePolicy>();
 
+// Avoid importing the proto implementation to reduce workflow bundle size
+// Copied from temporal.api.enums.v1.WorkflowIdConflictPolicy
+/**
+ * Defines what should happen when trying to start a Workflow with the same Workflow Id of a Running workflow.
+ *
+ * See `WorkflowIdReusePolicy` for handling Workflow Id duplication of a *Closed* workflow.
+ *
+ * *Note: It is never valid to have two actively running instances with the same Workflow Id.*
+ */
+export enum WorkflowIdConflictPolicy {
+  /**
+   *  This is the default so that we can set the policy
+   * `WORKFLOW_ID_REUSE_POLICY_TERMINATE_IF_RUNNING` in `WorkflowIdReusePolicy`, which is incompatible with setting any other
+   *  `WorkflowIdConflictPolicy` values.
+   *
+   * The actual default behavior is `WORKFLOW_ID_CONFLICT_POLICY_FAIL` for a start Workflow request, and `WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING` for a signal with start Workflow request.
+   */
```

Comment:
Unfortunately, this is the only way users can know what the actual default is, there is a discussion in the Python PR why we need to make that explicit in comments, and why the default cannot be directly set, I just followed that...

Reply:
I understand and agree with the rationale, but I'm also uncomfortable with the current phrasing. It is trying to justify why we came up with that technical decision rather than simply stating what this value is.

What about something like this?

```
Don't explicitly specify a Workflow ID Conflict Policy.

The server will determine the effective behavior as follow:
1. If `workflowIdReusePolicy` is set to `TERMINATE_IF_RUNNING`,
   then behave the same as `WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING`;
2. If request is `StartWorkflow`, then behave the same as `WORKFLOW_ID_CONFLICT_POLICY_FAIL`;
3. If request is `SignalWithStartWorkflow`, then behave the same as `WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING`.

Note: this is the only acceptable value for Conflict Policy if the Reuse Policy is set to `TERMINATE_IF_RUNNING`.
```

Reply:
Combining removal of the `WORKFLOW_ID_CONFLICT_POLICY_` prefix AND removal of the `UNSPECIFIED` value, you can update the `checkExtends` assertions to something like this:

```
checkExtends<
  `WORKFLOW_ID_CONFLICT_POLICY_${(keyof typeof WorkflowIdConflictPolicy) | 'UNSPECIFIED'}`,
  keyof typeof temporal.api.enums.v1.WorkflowIdConflictPolicy
>();
checkExtends<
  keyof typeof temporal.api.enums.v1.WorkflowIdConflictPolicy,
  `WORKFLOW_ID_CONFLICT_POLICY_${(keyof typeof WorkflowIdConflictPolicy) | 'UNSPECIFIED'}`
>();
```

And look at what we did for either `WorkflowUpdateStage` or `VersioningIntent` to convert those public enum values to the corresponding protobuf value (they use two distinct approaches, both are valid).

Reply:
I think the challenge is that we have  `WORKFLOW_ID_REUSE_POLICY_UNSPECIFIED`, again an optional field, and we could have got rid of the prefix. It is a bit too late for that without creating major deprecation issues. And we typically specify both reuse_policy and conflict_policy together, so it looks very confusing if one has the prefix, the other does not, and one can be set with an UNSPECIFIED field and the other cannot...

Reply:
So in that sense, if we were starting from scratch, I completely agree with you. But I think for consistency we need to keep the prefix and the unspecified field...